aptos/contracts/sources/governance/governance.move

```diff
@@ -0,0 +1,55 @@
+module pyth::governance {
+    use wormhole::vaa::{Self, VAA};
+    use pyth::data_source;
+    use wormhole::u16;
+    use pyth::governance_instruction;
+    use pyth::governance_action;
+    use pyth::contract_upgrade;
+    use pyth::set_governance_data_source;
+    use pyth::set_data_sources;
+    use pyth::set_stale_price_threshold;
+    use pyth::error;
+    use pyth::set_update_fee;
+    use pyth::state;
+
+    public entry fun execute_governance_instruction(vaa_bytes: vector<u8>) {
+        let parsed_vaa = parse_and_verify_governance_vaa(vaa_bytes);
+        let instruction = governance_instruction::from_byte_vec(vaa::destroy(parsed_vaa));
+
+        // Dispatch the instruction to the appropiate handler
+        let action = governance_instruction::get_action(&instruction);
+        if (action == governance_action::destroy(governance_action::new_contract_upgrade())) {
+            contract_upgrade::execute(governance_instruction::destroy(instruction));
+        } else if (action == governance_action::destroy(governance_action::new_set_governance_data_source())) {
+            set_governance_data_source::execute(governance_instruction::destroy(instruction));
+        } else if (action == governance_action::destroy(governance_action::new_set_data_sources())) {
+            set_data_sources::execute(governance_instruction::destroy(instruction));
+        } else if (action == governance_action::destroy(governance_action::new_set_update_fee())) {
+            set_update_fee::execute(governance_instruction::destroy(instruction));
+        } else if (action == governance_action::destroy(governance_action::new_set_stale_price_threshold())) {
```

Comment:
seems a little annoying that you have to call `::destroy` here to use the `Action` struct. Seems like you're not getting a lot of value from `Action` and it would be simpler / less code to use a `u8`. (or maybe you can use a typedef on u8?)